test/integration/src/peggy2/test_sifnode_transfers.py

```diff
@@ -0,0 +1,155 @@
+import time
+
+from siftool import eth, test_utils, sifchain
+from siftool.common import *
+import web3
+from json_file_writer import write_registry_json, ibc_json_file_path, double_peggy_json_file_path
+
+fund_amount_eth = eth.ETH
+rowan_unit = 10 ** 18
+fund_amount_sif = 2 * rowan_unit
+rowan_contract_address = web3.Web3.toChecksumAddress('0x5fbdb2315678afecb367f032d93f642f64180aa3')
+ibc_token_symbol = 'ibc/FEEDFACEFEEDFACEFEEDFACEFEEDFACEFEEDFACEFEEDFACEFEEDFACEFEEDFACE'
+double_peggy_symbol = 'sifBridge00030x1111111111111111111111111111111111111111'
+double_peggy_address = '0x1111111111111111111111111111111111111111'
+
+fund_amount_ibc = 10000
+fund_amount_double_peggy = 10000
+
+admin_account = "sifnodeadmin"
+
+def test_rowan_to_eth_and_back_to_sifnode_transfer_valid(ctx):
+    # get rowan contract
+    rowan_sc = ctx.get_generic_erc20_sc(rowan_contract_address)
+
+    # Create/retrieve a test ethereum account
+    test_eth_account = ctx.create_and_fund_eth_account(fund_amount=fund_amount_eth)
+
+    # create/retrieve a test sifchain account
+    test_sif_account = ctx.create_sifchain_addr(fund_amounts=[[fund_amount_sif, "rowan"], [fund_amount_eth, ctx.ceth_symbol]])
+
+    # init balance for erc20 rowan
+    test_eth_account_initial_balance = ctx.get_erc20_token_balance(rowan_contract_address, test_eth_account)
+
+    # sif account initial balance
+    test_sif_account_initial_balance = ctx.get_sifchain_balance(test_sif_account)
+
+    # lock rowan in sifnode
+    amount_to_lock = 1 * rowan_unit
+    ctx.sifnode_client.send_from_sifchain_to_ethereum(test_sif_account, test_eth_account, amount_to_lock, "rowan")
+    ctx.advance_blocks()
+    ctx.wait_for_sif_balance_change(test_sif_account, test_sif_account_initial_balance, [[amount_to_lock, "rowan"]])
+
+    # wait rowan change happened in sifnode account
+    test_sif_account_after_lock_balance = ctx.get_sifchain_balance(test_sif_account)
+
+    # we need take the transaction fee into account
+    rowan_balance = test_sif_account_initial_balance["rowan"] - amount_to_lock
+    assert rowan_balance >= test_sif_account_after_lock_balance["rowan"]
+
+    # wait the ethereum reciever's rowan balance change
+    ctx.wait_for_eth_balance_change(test_eth_account, test_eth_account_initial_balance, token_addr=rowan_contract_address, timeout=90)
```

Comment:
Nit: This function returns the delta. No need to fetch balance again in next line

Reply:
got it, thanks